Added basic JWT context with example.

rakshith-ravi opened the pull request:
No clue if it works or not.

@trezm I'm not even sure if I understand the way this is supposed to work. Do let me know if you want me to make any other changes.

trezm commented:
Hey there @rakshith-ravi, this doesn't compile for me right now -- missing the `frank_jwt` dep and a few imports.

Beyond that, I'd recommend moving the custom context you've made into its own file, and then putting the whole thing in a `jwt_example` folder with a `main.rs`. That will help logically separate the code for newcomers looking at this example as well as keep the code a little cleaner :)

rakshith-ravi commented:
Updated my PR. The example doesn't seem to compile for some reason. Fixed all other issues. However, it keeps expecting a `BasicContext` when I've clearly initialized it as a `JwtContext`. Could you take a look at it and see if I'm missing something?

trezm commented on thruster/examples/jwt_example/jwt_context.rs:
```diff
@@ -1,18 +1,15 @@
+    Context, Request, Response,
+};
+
+pub struct JwtContext {
```
If you want to use a custom context, you must implement the `Context` trait, you can find that [here](https://github.com/trezm/Thruster/blob/master/thruster/src/core/context.rs).

rakshith-ravi replied:
Wait, but I am implementing `Context` though. Check out line number 134. I'm implementing `Context` for `JwtContext`

trezm replied:
My fault, for some reason the diff I was looking at didn't have passed line 21!

rakshith-ravi replied:
No worries. Don't merge this yet. I'm noticing some issue on my side. I'll let you know once it's fixedx

trezm commented on thruster/examples/jwt_example/main.rs:
```diff
@@ -0,0 +1,124 @@
+    env_logger::init();
+    info!("Starting server...");
+
+    let mut app = App::<Request, JwtContext>::new_basic();
```
You'll also need to use a different method for creating the app, one that takes a context generator as an argument, so you'll need something like:

```rust
pub fn generate_context(request: Request) -> JwtContext {
    let mut ctx = JwtContext::new();
    ctx.params = request.params().clone();
    ctx.request = request;

    ctx
}

let mut app = App::<Request, JwtContext>:: create(generate_context);
```

rakshith-ravi commented:
Okay yeah. Made the changes. It compiles well now.

rakshith-ravi commented:
Okay, yeah. All fixed and working

trezm commented:
Awesome, thanks so much for the contribution!! All merged in!
